tests/checkpointing/test_model_checkpoint.py

```diff
@@ -1202,3 +1202,37 @@ def test_check_val_every_n_epochs_top_k_integration(tmpdir):
     )
     trainer.fit(model)
     assert set(os.listdir(tmpdir)) == {"epoch=1.ckpt", "epoch=3.ckpt"}
+
+
+def test_model_checkpoint_loadsave_ckpt(tmpdir):
```

Comment:
shouldn't the test first save then load? 
Also, it would be best if you re-instantiate the modelcheckpoint callback for each call.

Reply:
> shouldn't the test first save then load? Also, it would be best if we re-instantiate the modelcheckpoint callback for each call.

Since these are pure unit tests the sequence strictly doesn't really matter that much (but I guess makes them easier to follow).

I applied your suggestions and clarified/documented that we currently don't restore all states that are written to disk.